core/src/replay_stage.rs

```diff
@@ -3574,19 +3615,25 @@ impl ReplayStage {
                 && propagation_confirmed
                 && switch_fork_decision.can_vote()
             {
-                info!("voting: {} {}", bank.slot(), fork_weight);
+                info!("voting: {} {}", candidate_vote_bank.slot(), fork_weight);
                 SelectVoteAndResetForkResult {
-                    vote_bank: Some((bank.clone(), switch_fork_decision)),
-                    reset_bank: Some(bank.clone()),
+                    vote_bank: Some((candidate_vote_bank.clone(), switch_fork_decision)),
```

Comment:
nit: maybe a comment or assert saying if we're here than candidate_vote_bank == reset_bank always

Reply:
done, also replaced `candidate_vote_bank` with `heaviest_bank_on_same_voted_fork` before the decision is made for clarity